Dockerfile

```diff
@@ -0,0 +1,75 @@
+# Dockerfile to build and serve datagateway
+
+# Build stage
+FROM node:16.14-alpine3.15 as builder
```

Comment:
Node 16 is now no longer supported, Node 20 is current LTS and is what we're now running CI on.

I forgot to mention this for the SG PR, but can we test these containers with the up-to-date LTS version of Node?

Reply:
```suggestion
FROM node:20.9-alpine3.18 as builder
```

Reply:
I updated it to Node 20 in https://github.com/ral-facilities/datagateway/pull/1535/commits/f9c87e00c8019a4deb8f900d069de38874cd6186 and it seems to be working fine. Feel free to test it.